inc/ThirdParty/Plugins/Ecommerce/XLWooCommerceSalesTriggersSubscriber.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace WP_Rocket\ThirdParty\Plugins\Ecommerce;
+
+use WP_Rocket\Engine\Optimization\DelayJS\HTML;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+class XLWooCommerceSalesTriggersSubscriber implements Subscriber_Interface {
+
+	/**
+	 * Delay JS HTML class.
+	 *
+	 * @var HTML
+	 */
+	private $delayjs_html;
+
+	/**
+	 * WooCommerceSubscriber constructor.
+	 *
+	 * @param HTML $delayjs_html DelayJS HTML class.
+	 */
+	public function __construct( HTML $delayjs_html ) {
+		$this->delayjs_html = $delayjs_html;
+	}
+
+	/**
+	 * Get the subscribed events array.
+	 *
+	 * @return array|array[]
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'rocket_delay_js_exclusions' => [ 'modify_delayjs_exclusions_on_product_pages', 10, 2 ],
+		];
+	}
+
+	/**
+	 * Modify delay JS excluded scripts for compatibility.
+	 *
+	 * @param array $exclusions The initial Delay JS exclusion list.
+	 *
+	 * @return array The filtered Delay JS exclusion list.
+	 */
+	public function modify_delayjs_exclusions_on_product_pages( array $exclusions ) {
+		if ( empty( rocket_get_constant( 'WCST_VERSION' ) ) ) {
+			return $exclusions;
+		}
+
+		if ( 'product' !== get_post_type() || ! is_single() ) {
```

Comment:
In the WC context, you can use `is_product()` directly instead

Reply:
Good point.